aim/sdk/adapters/xgboost.py

```diff
@@ -0,0 +1,45 @@
+from xgboost.callback import TrainingCallback, CallbackContainer
```

Comment:
lets implement an interface like PL, huggingface integrations to not have direct dependency from xgboost

Reply:
e.g.
```
class AimCallback(object):
    @staticmethod
    def __new__(cls, *args, **kwargs):
        xgboost_logger_inst = cls.__get_logger_cls()
        return xgboost_logger_inst(*args, **kwargs)

    @classmethod
    def __get_logger_cls(cls):
        ...
        from xgboost.callback import TrainingCallback
        class _XGBoostLogger(TrainigCallback):
            ...
```